tests/unit/rpc/handlers/LedgerEntryTests.cpp

```diff
@@ -2612,3 +2614,293 @@ TEST(RPCLedgerEntrySpecTest, DeprecatedFields)
+        "ledger_index":30,
+        "validated":true,
+        "index":"05FB0EB4B899F056FA095537C5817163801F544BAFCEA39C995D76DB4D16F9DD",
+        "deleted_ledger_index":30,
+        "node":{
+            "Amount":"123",
+            "Flags":0,
+            "LedgerEntryType":"NFTokenOffer",
+            "NFTokenID":"00010000A7CAD27B688D14BA1A9FA5366554D6ADCF9CE0875B974D9F00000004",
+            "NFTokenOfferNode":"0",
+            "Owner":"rf1BiGeXwwQoi8Z2ueFYTEXSwuJYfV2Jpn",
+            "OwnerNode":"0",
+            "PreviousTxnID":"0000000000000000000000000000000000000000000000000000000000000000",
+            "PreviousTxnLgrSeq":0,
+            "index":"05FB0EB4B899F056FA095537C5817163801F544BAFCEA39C995D76DB4D16F9DD"
+            }
+        })";
+    backend->setRange(RANGEMIN, RANGEMAX);
+    auto const ledgerinfo = CreateLedgerInfo(LEDGERHASH, RANGEMAX);
+    EXPECT_CALL(*backend, fetchLedgerBySequence(RANGEMAX, _)).WillRepeatedly(Return(ledgerinfo));
+    // return valid ledger entry which can be deserialized
+    auto const offer = CreateNFTBuyOffer(NFTID, ACCOUNT);
+    EXPECT_CALL(*backend, doFetchLastTwoLedgerObjects(ripple::uint256{INDEX1}, RANGEMAX, _))
+        .WillRepeatedly(Return(std::vector<std::pair<std::uint32_t, Blob>>{
+            {ledgerinfo.seq, {}},
+            {ledgerinfo.seq - 1, offer.getSerializer().peekData()}
+        }));
+
+    runSpawn([&, this](auto yield) {
+        auto const handler = AnyHandler{LedgerEntryHandler{backend}};
+        auto const req = json::parse(fmt::format(
+            R"({{
+                "index": "{}",
+                "include_deleted": true
+            }})",
+            INDEX1
+        ));
+        auto const output = handler.process(req, Context{yield});
+        ASSERT_TRUE(output);
+        EXPECT_EQ(*output.result, json::parse(OUT));
+    });
+}
+
+TEST_F(RPCLedgerEntryTest, LedgerEntryNotExist)
+{
+    auto const ledgerinfo = CreateLedgerInfo(LEDGERHASH, RANGEMAX);
+    EXPECT_CALL(*backend, fetchLedgerBySequence(RANGEMAX, _)).WillRepeatedly(Return(ledgerinfo));
+
+    auto const offer = CreateNFTBuyOffer(NFTID, ACCOUNT);
+    EXPECT_CALL(*backend, doFetchLastTwoLedgerObjects(ripple::uint256{INDEX1}, RANGEMAX, _))
+        .WillRepeatedly(Return(std::vector<std::pair<std::uint32_t, Blob>>{
+        }));
+
+    runSpawn([&, this](auto yield) {
+        auto const handler = AnyHandler{LedgerEntryHandler{backend}};
+        auto const req = json::parse(fmt::format(
+            R"({{
+                "index": "{}",
+                "include_deleted": true
+            }})",
+            INDEX1
+        ));
+        auto const output = handler.process(req, Context{yield});
+        ASSERT_FALSE(output);
+        auto const err = rpc::makeError(output.result.error());
+        auto const myerr = err.at("error").as_string();
+        EXPECT_EQ(myerr, "entryNotFound");
+    });
+}
+
+TEST_F(RPCLedgerEntryTest, LedgerEntryNotExist2)
```

Comment:
how is this test different from the previous test `LedgerEntryNotExist`?

Reply:
The first one is the LastTwoObject is empty. From code level it will be catch by lastTwoObjects.empty(). This one is possible in real world. (https://github.com/Kassaking7/clio/blob/project-DEFI-129-branch/src/rpc/handlers/LedgerEntry.cpp#L165)

This one is ledgerObject is empty. So from code level it will be catch by if (!ledgerObject || ledgerObject->empty()). This one is usually not possible in real world. (https://github.com/Kassaking7/clio/blob/project-DEFI-129-branch/src/rpc/handlers/LedgerEntry.cpp#L170)

Added this one to cover all if-else statements in my code

Reply:
could we expect the return value as `std::vector<std::pair<std::uint32_t, Blob>>{ {0, Blob{}} }` instead?

This is easier to read and we would want readability over simplicity sometimes

Reply:
Oh yeah, it looks better! Let me make some change on this one